src/test/java/org/kiwiproject/champagne/resource/DeploymentEnvironmentResourceTest.java

```diff
@@ -0,0 +1,325 @@
+package org.kiwiproject.champagne.resource;
+
+import static javax.ws.rs.client.Entity.json;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.kiwiproject.collect.KiwiLists.first;
+import static org.kiwiproject.test.jaxrs.JaxrsTestHelper.assertAcceptedResponse;
+import static org.kiwiproject.test.jaxrs.JaxrsTestHelper.assertInternalServerErrorResponse;
+import static org.kiwiproject.test.jaxrs.JaxrsTestHelper.assertNoContentResponse;
+import static org.kiwiproject.test.jaxrs.JaxrsTestHelper.assertOkResponse;
+import static org.kiwiproject.test.jaxrs.JaxrsTestHelper.assertResponseStatusCode;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.kiwiproject.champagne.dao.AuditRecordDao;
+import org.kiwiproject.champagne.dao.DeploymentEnvironmentDao;
+import org.kiwiproject.champagne.model.AuditRecord;
+import org.kiwiproject.champagne.model.DeploymentEnvironment;
+import org.kiwiproject.champagne.model.AuditRecord.Action;
+import org.kiwiproject.champagne.util.AuthHelper;
+import org.kiwiproject.jaxrs.exception.JaxrsExceptionMapper;
+import org.mockito.ArgumentCaptor;
+
+import io.dropwizard.testing.junit5.DropwizardExtensionsSupport;
+import io.dropwizard.testing.junit5.ResourceExtension;
+
+import javax.ws.rs.core.GenericType;
+
+@DisplayName("DeploymentEnvironmentResource")
+@ExtendWith( DropwizardExtensionsSupport.class)
+class DeploymentEnvironmentResourceTest {
+    private static final DeploymentEnvironmentDao DEPLOYMENT_ENVIRONMENT_DAO = mock(DeploymentEnvironmentDao.class);
+    private static final AuditRecordDao AUDIT_RECORD_DAO = mock(AuditRecordDao.class);
+    private static final DeploymentEnvironmentResource DEPLOYMENT_ENVIRONMENT_RESOURCE = new DeploymentEnvironmentResource(DEPLOYMENT_ENVIRONMENT_DAO, AUDIT_RECORD_DAO);
+
+    private static final ResourceExtension APP = ResourceExtension.builder()
+            .bootstrapLogging(false)
+            .addResource(DEPLOYMENT_ENVIRONMENT_RESOURCE)
+            .addProvider(JaxrsExceptionMapper.class)
+            .build();
+    
+    @BeforeEach
+    void setUp() {
+        reset(DEPLOYMENT_ENVIRONMENT_DAO, AUDIT_RECORD_DAO);
+
+        AuthHelper.setupCurrentPrincipalFor("bob");
```

Comment:
For later, you might as well create a simple Jupiter extension to do this before & after each test. Since you're going to need it in each resource test...